### PR TITLE
Feature/seqng 1043

### PR DESCRIPTION
**Pull request by jluhrs**
All systems' EPICS statuses are now read as `F[A]` instead of `F[Option[A]]`, leaving to `F` the task of dealing with null values.

### CODE REVIEW

File: modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala [hunk 1]
```diff
@@ -232,17 +232,17 @@ object EpicsUtil {
...
+      .ensure(NullEpicsError(channel))(_.isDefined) // equivalent to a null check
       .map{_.orNull} // orNull lets us typecheck but it will never be used due to the `ensure` call above
 
   def safeAttributeF[F[_]: Sync, A >: Null](get: => CaAttribute[A]): F[A] =
-    safeAttributeWrapF(get.name, get.value)
+    safeAttributeWrapF(get.channel, get.value)
 
   def safeAttributeSListF[F[_]: Sync, A >: Null](get: => CaAttribute[A]): F[List[A]] =
-    safeAttributeWrapF(get.name, get.values.asScala.toList)
+    safeAttributeWrapF(get.channel, get.values.asScala.toList)
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/EpicsUtil.scala]: Having the channel address is more useful for debugging.

File: modules/seqexec/server/src/main/scala/seqexec/server/altair/AltairKeywordReader.scala [hunk 2]
```diff
@@ -82,14 +81,12 @@ object AltairKeywordReaderEpics extends AltairKeywordReaderLUT {
   def apply[F[_]: Sync](sys: AltairEpics[F]): AltairKeywordReader[F] = new AltairKeywordReader[F] {
 
     override def aofreq: F[Double] =
-      Nested(sys.aoexpt)
-        .filter(_ =!= 0.0f)
-        .map(1 / _.toDouble)
-        .value
+      sys.aoexpt
+        .map(1.0 / _.toDouble)
```
Comment (cquiroz) [inline modules/seqexec/server/src/main/scala/seqexec/server/altair/AltairKeywordReader.scala]: Why is the filter off? Are you assuming that `safeValOrDefalt` would catch a divide by zero?

File: modules/seqexec/server/src/main/scala/seqexec/server/gcal/GcalKeywordReader.scala [hunk 3]
```diff
@@ -41,25 +41,25 @@ object GcalKeywordsReaderEpics {
...
-      val ir   = Nested(sys.lampIr.delay).map{
-        case BinaryOnOff.ON => "IRhigh"
-        case _              => "IRlow"
-      }.value
-      val qh   = sys.lampQH.delay.map(_.exists(onCheck)).ifM("QH".some.pure[F], noValue)
-      val thAr = sys.lampThAr.delay.map(_.exists(onCheck)).ifM("ThAr".some.pure[F], noValue)
-      val xe   = sys.lampXe.delay.map(_.exists(onCheck)).ifM("Xe".some.pure[F], noValue)
+      def onCheck(v: BinaryOnOff): Boolean = v === BinaryOnOff.ON
 
-      ar.orElse(xe).orElse(cuAr).orElse(thAr).orElse(qh).orElse(ir).safeValOrDefault
```
Comment (jluhrs) [inline modules/seqexec/server/src/main/scala/seqexec/server/gcal/GcalKeywordReader.scala]: This shouldn't work. The intention is that the result must be the first `Some(_)`, or a default value if none. But the `orElse` here is run over `F`, not teh contained `Option`!

File: modules/seqexec/server/src/main/scala/seqexec/server/altair/AltairKeywordReader.scala (hunk 2 above)
Comment (jluhrs) [reply modules/seqexec/server/src/main/scala/seqexec/server/altair/AltairKeywordReader.scala]: Yes

**Pull request merged**